Number text field

ParimSsme opened the pull request.

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

ParimSsme committed:
add UstadNumberTextEditField.kt

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/composable/UstadNumberTextEditField.kt:
```diff
@@ -0,0 +1,100 @@
+                imeAction = ImeAction.Done
+            ),
+            keyboardActions = KeyboardActions(
+                onDone = {
```
This will only run if the user clicks the done checkbox on the keyboard, I suggest removing min/max. Let this be handled by the viewmodel

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/composable/UstadNumberTextEditField.kt:
```diff
@@ -0,0 +1,100 @@
+
+@Composable
+fun UstadNumberTextEditField(
+    value: String,
```
The value of the number field won't be a string, it would be a number.

ParimSsme committed:
made changes from pull request comments

ParimSsme committed:
made changes from pull request comments

ParimSsme committed:
made changes from pull request comments

ParimSsme committed:
made changes from pull request comments

mikedawson committed:
Update UstadNumberTextField to display only digits. Make argument types for label and placeholder match Jetpack compose function itself.

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

ParimSsme committed:
add UstadNumberTextEditField.kt in web

mikedawson committed:
Adjust virtual list and make it possible to create a grid using paged data.

mikedawson committed:
Implement ClazzList using ViewModel.

mikedawson committed:
Remove ClazzListPresenter etc.

mikedawson committed:
Initial basic implementation of ClazzEditViewModel.

mikedawson committed:
Use ClazzEditViewModel on React ClazzEditScreen. Fix Clazz equals/hashcode function. Refactor MessageIdSelect into its own file.

mikedawson committed:
Remove use of old Klock library from all modules that compile to Javascript (to facilitate move to IR compiler).

mikedawson committed:
Switch to using JS IR compiler.

mikedawson committed:
Update Kotlin JS wrappers, including update to new react-router package.

mikedawson committed:
Update Kotlin JS wrappers to pre.526

mikedawson committed:
Improve React DatePicker user experience. Switch to using native date input type.

mikedawson committed:
Improve React DatePicker user experience. Switch to using native date input type.

mikedawson committed:
Improve implementation of ClazzEditViewModel to start handling sub editing (e.g. editing schedule etc). Add ScheduleEdit ViewModel and tests.

mikedawson committed:
Fix NavHost on Javascript to ensure children are shown after finishing popping when there is no additional navigation to do.
Add support for NavResultReturner on React.
Replace inline JSON functions with use of KSerializer
Fix Schedule display on React (need to use own component - avoid use function within loop)

mikedawson committed:
Make get/set Json in UstadViewModel saved state handle use a suspended function to avoid this running on the main thread on Android.
Fix load/commit for schedules on ClazzEdit.
Remove view xml and old schedule edit class component.

mikedawson committed:
Add onTextChange property for TextField on React. ClazzEditViewModel function to handle when user clicks on add a new block.

mikedawson committed:
Work in progress adding functional  component wrapper for DraftJS

mikedawson committed:
Remove class component based Javascript

mikedawson committed:
Remove usage of moment js. Allow use of webpack 5 (previously disabled due to Kotlin/JS bug, since resolved). Fix webpack fallbacks for asmcrypto module.

mikedawson committed:
Add Quill CSS. Remove use of DraftJS.

mikedawson committed:
Update Android Gradle plugin 7.4.0 to 7.4.2

mikedawson committed:
HTML rich text editing on Android: move HTML rich text editing to a separate fragment (needed to have sufficient screen space).
Move pbkdf encryption onto server on Javascript: previous function required untyped js that no longer works after updating webpack.
Add id on various web elements.
Update clickable text field on Android to avoid triggering the click function when the user is scrolling.
Update CourseBlockEdit to add title and description editing
Fix NavResultReturner issue where a result from a previous sceren could be accidentally collected

mikedawson committed:
Remove import mui.icons.material from Javascript - this was making autocomplete unusably slow.

mikedawson committed:
Add implementation for moving, indenting, hiding, reordering course blocks.

mikedawson committed:
Merge clazzvm

mikedawson committed:
Fix implementation of UstadNumberTextField: handle prop changes with useEffect. Make prop class extend textfield props.

mikedawson committed:
Change number text field to use float type so that floating point can be used in future when required without the need for a lot of refactoring.
